File: src/backend/NHP_Domain/Entities/Prestation.cs
namespace NHP_Domain.Entities;

public class Prestation
{
    public long Id { get; set; }
    public String NomPrestation { get; set; }
    
    // OneToMany
    public virtual List<PrestationProposee> PrestationProposees { get; set; }

    public override string ToString()
    {
        return $"nomPrestation: {NomPrestation}";
    }
}

File: src/frontend/src/domain/entities/Prestation.ts
import { type IPrestationProposee } from '@/domain/entities/PrestationProposee';

export interface IPrestation {
    id: number;
    nomPrestation: string;
    prestationProposees: IPrestationProposee[];
}

export class Prestation implements IPrestation {
    constructor(
        public id: number,
        public nomPrestation: string,
        public prestationProposees: IPrestationProposee[]
    ) {}
}